chore: merge in starbase (#163)
canonical/craft-archives: README.md

# Craft Archives

[![Documentation Status][rtd-badge]][rtd-latest]
[![Codecov Status][codecov-badge]][codecov-status]
[![Ruff status][ruff-badge]][ruff-site]

> Craft Archives is a Python package to manage interaction with software archives on behalf of tools using the Craft Parts framework.

Reviewer: ```suggestion
Craft Archives is a package for tools that use the Craft Parts framework. It manages their interactions with software archives.
```

- Do we officially consider Craft Parts a framework?
- We might want to be precise about what "software archives" means.

Reviewer: The nomenclature points to "[software repository](https://en.wikipedia.org/wiki/Software_repository)", as much as we might confuse that with a Git repo. "Software package repository" could be even more precise.

"Archive" in Debian land refers to "[old distributions of Debian](https://www.debian.org/distrib/archive)".

Author: I like "software package repository" a lot - it's clear and disambiguates from Git as well. The term comes up pretty often under that one header, how do you feel about calling it "software package repositories" once and then "repositories" onwards?

Reviewer: > how do you feel about calling it "software package repositories" once and then "repositories" onwards?

:+1:

> Craft Archives provides a set of common interfaces to define the locations of software archives, install packages and verify their integrity.

Reviewer: Install packages where?

Author: This language came from the existing README, but I think it's actually wrong. Craft Archives appears to just add support for the `package-repositories` key in project files, while Craft Parts does the actual downloading of files. I'll remove this language

> Craft Archives provides a set of common interfaces to define the locations of software archives, install packages and verify their integrity.

Reviewer: ```suggestion
Craft Archives provides a set of common interfaces to define the locations of software archives, install packages, and verify package integrity.
```

Reviewer: Oh oops, I was looking at `docs/index.rst`, which has the old text.

> The aim of this package is to provide a uniform, extensible set of interfaces that other tools and packages can use to work with software archives.

Reviewer: ```suggestion
It aims to provide a uniform, extensible set of interfaces that other tools and packages can use when working with software archives.
```

> This package is most useful for implementers of tools using the Craft Parts framework that need to provide support for additional software archives.

Reviewer: ```suggestion
This package is most useful for developers who write tools with Craft Parts and need to provide support for additional software archives.
```

# Craft Archives

[![Documentation Status][rtd-badge]][rtd-latest]
[![Codecov Status][codecov-badge]][codecov-status]
[![Ruff status][ruff-badge]][ruff-site]

Craft Archives is a Python package to manage interaction with software archives on behalf of tools using the [Craft Parts] library.

Craft Archives provides a set of common interfaces to define the locations of software archives and install new software repositories.

It aims to provide a uniform, extensible set of interfaces that other tools and packages can use when working with software archives.

> This package is most useful for developers who write tools with [Craft Parts] and need to provide support for additional software archives.

Reviewer: Missing line wrapping on these.
